### PR TITLE
Parameter Checking

### PR DESCRIPTION
**Pull request by matthewrsj**
#44

### CODE REVIEW

**Comment by matthewrsj** [comment]
I also made the tests check that errors are a list containing an error python dictionary. This is easy to implement here.

**Comment by Kennric** [comment]
Tests pass, looks good. +1

**Comment by Kennric** [comment]
(pending possible refactor of string returns for check_params)

**Comment by matthewrsj** [comment]
I like @Kennric's suggestion on IRC: get_field_errors returning None or a string reporting the bad field is more intuitive

I will implement

File: pymesync.py [hunk 1]
```diff
@@ -247,3 +275,30 @@ def _format_endpoints(self, queries):
             query_string += "?{}".format("&".join(query_list))
 
         return query_string
+
+    def _get_field_errors(self, actual, object_name):
+        """Checks that ``actual`` parameter passed to POST method contains
+        items in required or optional lists for that ``object_name``.
+        Returns None if no errors found or error string if error found"""
+        missing_list = list(self.required_params[object_name])
```
Comment (Kennric) [inline pymesync.py]: This is a very minor nitpick, but you don't need to bother assigning this variable until you are sure the next test passes - check that you have a dict first, then do everything else.

**Comment by Kennric** [comment]
Looks good +1

**Pull request merged**